docs/src/test/scala-2.12+/docs/http/scaladsl/HttpServerWithTypedSample.scala

```diff
@@ -0,0 +1,228 @@
+      def write(status: Status): JsValue = status match {
+        case Failed     => JsString("Failed")
+        case Successful => JsString("Successful")
+      }
+
+      def read(json: JsValue): Status = json match {
+        case JsString("Failed")     => Failed
+        case JsString("Successful") => Successful
+        case _                      => throw new DeserializationException("Status unexpected")
+      }
+    }
+
+    implicit val jobFormat = jsonFormat4(Job)
+
+    implicit val jobsFormat = jsonFormat1(Jobs)
+  }
+  //#akka-typed-json
+
+  //#akka-typed-route
+  import akka.actor.typed.ActorSystem
+  import akka.util.Timeout
+
+  import akka.http.scaladsl.server.Directives._
+  import akka.http.scaladsl.model.StatusCodes
+  import akka.http.scaladsl.server.Route
+
+  import scala.concurrent.duration._
+  import scala.concurrent.Future
+
+  class JobRoutes(buildJobRepository: ActorRef[JobRepository.Command])(implicit system: ActorSystem[_]) extends JsonSupport {
+
+    import akka.actor.typed.scaladsl.AskPattern._
+
+    // asking someone requires a timeout and a scheduler, if the timeout hits without response
+    // the ask is failed with a TimeoutException
+    implicit val timeout: Timeout = 3.seconds
+    // implicit scheduler only needed in 2.5
+    // in 2.6 having an implicit typed ActorSystem in scope is enough
+    implicit val scheduler = system.scheduler
+
+    lazy val theJobRoutes: Route =
+      pathPrefix("jobs") {
+        concat(
+          pathEnd {
+            concat(
+              post {
+                entity(as[JobRepository.Job]) { job =>
+                  val operationPerformed: Future[JobRepository.Response] =
+                    buildJobRepository.ask(JobRepository.AddJob(job, _))
+                  onSuccess(operationPerformed) {
+                    case JobRepository.OK         => complete("Job added")
+                    case JobRepository.KO(reason) => complete(StatusCodes.InternalServerError -> reason)
+                  }
+                }
+              },
+              delete {
+                val operationPerformed: Future[JobRepository.Response] =
+                  buildJobRepository.ask(JobRepository.ClearJobs(_))
+                onSuccess(operationPerformed) {
+                  case JobRepository.OK         => complete("Jobs cleared")
+                  case JobRepository.KO(reason) => complete(StatusCodes.InternalServerError -> reason)
+                }
+              }
+            )
+          },
+          (get & path(LongNumber)) { id =>
+            val maybeJob: Future[Option[JobRepository.Job]] =
+              buildJobRepository.ask(JobRepository.GetJobById(id, _))
+            rejectEmptyResponse {
+              complete(maybeJob)
+            }
+          }
+        )
+      }
+  }
+  //#akka-typed-route
+
+  /* there are still differences making this impossible to compile both for 2.5 and 2.6 at the same time
```

Comment:
So, for now, this would compile correctly in the regular build, but would fail building against 2.6 master.